include/TcpServer.h

```diff
@@ -12,6 +14,9 @@ class TcpServer {
 public:
     TcpServer(void* (*workerThreadProc)(void*));
     ~TcpServer();
+    void listen(const char* host, int port);
+    void accept(const std::atomic_bool& terminate_flag);
+    virtual void addHandler(HttpMethod method, const char* regexpStr, HttpRequestHandler handler) = 0;
```

Comment:
addHandler is an HTTP-specific method. It doesn't make any sense on this abstraction level.